src/m365/pp/commands/managementapp/managementapp-add.ts

```diff
@@ -0,0 +1,132 @@
+import { Application } from '@microsoft/microsoft-graph-types';
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+export interface Options extends GlobalOptions {
+  appId?: string;
+  objectId?: string;
+  name?: string;
+}
+
+class PpManagementAppAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MANAGEMENTAPP_ADD;
+  }
+
+  public get description(): string {
+    return 'Register management application for Power Platform';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.objectId = typeof args.options.objectId !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getAppObjectId(args)
+      .then((appId: string): Promise<any> => {
```

Comment:
Let's rename `appId` to `objectId` to clearly indicate what it is

Reply:
The method name that is called is is wrong; we are not using the `objectId` but the `appId`. I renamed the method to reflect this.